src/main/java/net/dries007/tfc/common/recipes/SimplePotRecipe.java

```diff
@@ -0,0 +1,239 @@
+package net.dries007.tfc.common.recipes;
+
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Queue;
+import javax.annotation.Nullable;
+
+import com.google.gson.JsonObject;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.item.crafting.ShapedRecipe;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.ListNBT;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.common.util.Constants;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.items.ItemStackHandler;
+
+import com.mojang.serialization.JsonOps;
+import net.dries007.tfc.util.Helpers;
+
+//todo: much of the functionality is here, just needs to be finished
+public class SimplePotRecipe implements IPotRecipe
+{
+    protected final ResourceLocation id;
+    protected final HashSet<Ingredient> inputItems;
+    protected final NonNullList<ItemStack> outputItems;
```

Comment:
Is there any reason why we use `NonNullList` here? I don't particularly like it as it's missing helpful functionality of `ArrayList` or `LinkedList`, and the copy into the latter kind of suggests that.